Connection/Connection.php

```diff
@@ -337,14 +337,33 @@ public function disconnect()
 
     /**
      * Set socket.
+     * A wrapper can be registered to handle application layer transport
      *
-     * @param   string  $socket    Socket URI.
+     * @param   string  $socketUri    Socket URI.
      * @return  \Hoa\Socket
+     * @throws  \Hoa\Socket\Exception
      */
-    protected function setSocket($socket)
-    {
+    protected function setSocket($socketUri)
+    {
+        $parsed = parse_url($socketUri);
+        if ($parsed !== false &&
+            isset($parsed['scheme']) &&
+            null !== $wrapper = Socket\Transport::getWrapper($parsed['scheme'])
+            ) {
+            $socket = $wrapper($socketUri);
```

Comment:
@Hywan, I tested and if the variable can be executed as a callable, the parenthesis worked... So I've updated to remove call_user_func here.

```php
class toto {
    public static function varT() {
        echo uniqid();
    }
    public function varP() {
        echo uniqid();
    }
}

$callable = ['toto', 'varT'];
var_dump($callable()); //Works

$t = new toto();
$callable = [$t, 'varP'];
var_dump($callable()); //Works

$callable = function() {echo uniqid();};
var_dump($callable()); //Works
```